.github/workflows/docker-sign.yml

```diff
@@ -0,0 +1,40 @@
+name: docker-sign-container
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    permissions:
+      id-token: write  # undocumented OIDC support.
+      packages: write
+      contents: read
+    env:
+      COSIGN_EXPERIMENTAL: "true"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
```

Comment:
### Pin actions to a full length commit SHA

Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.

https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
Pinning a specific SHA sounds good. Do you have a recommendation for keeping up to date in case of new versions? Seems like they may not change often, but it would be helpful to be notified when it happens. Do you typically add a check to the build scripts for each project, or watch the source repos directly and keep track of which repos use which actions elsewhere?

Reply:
Dependabot takes care of version upgrades.